dask_expr/tests/test_groupby.py

```diff
@@ -383,6 +383,25 @@ def test_groupby_median(df, pdf):
     assert_eq(df.groupby("x").median()["y"], pdf.groupby("x").median()["y"])
 
 
+def test_groupby_ffill_bfill(pdf):
+    pdf["y"] = pdf["y"].astype("float64")
+
+    pdf.iloc[np.arange(0, len(pdf) - 1, 3), 1] = np.nan
+    df = from_pandas(pdf, npartitions=10)
+    assert_eq(df.groupby("x").ffill(), pdf.groupby("x").ffill())
+    assert_eq(df.groupby("x").bfill(), pdf.groupby("x").bfill())
+
+    actual = df.groupby("x")["y"].ffill()
+    expect = df[["x", "y"]].groupby("x")["y"].ffill()
+    assert actual.optimize()._name == expect.optimize()._name
+    assert_eq(actual, pdf.groupby("x")["y"].ffill())
+
+    actual = df.groupby("x").ffill()["y"]
+    expect = df[["x", "y"]].groupby("x").ffill()["y"]
+    assert actual.optimize()._name == expect.optimize()._name
+    assert_eq(actual, pdf.groupby("x")["y"].ffill())
```

Comment:
I tried testing that `.optimize()._name` is the same across all of these:

- `df.groupby("x")["y"].ffill()`
- `df.groupby("x").ffill()["y"]`
- `df[["x", "y"]].groupby("x")["y"].ffill()`
- `df[["x", "y"]].groupby("x").ffill()["y"]`


but that's not the case. Is it expected?

Reply:
Yes this behaves a little different. The first caste reduces the groupby op to a series while the second does the groupby on a DataFrame with one column and reduces to a series in the end. 

Making this consistent is not worth the effort for now, since it is the same behavior and performance wise

Reply:
The slice is not pushed into groupby, which is the difference that you are seeing here